src/codeflare_sdk/ray/cluster/cluster.py

```diff
@@ -74,10 +72,16 @@ def __init__(self, config: ClusterConfiguration):
         request.
         """
         self.config = config
-        self.app_wrapper_yaml = self.create_app_wrapper()
+        if self.config is None:
+            warnings.warn(
+                "Please provide a ClusterConfiguration to initialise the Cluster object"
+            )
+            return
+        else:
+            self.resource_yaml = self.create_resource()
+
         self._job_submission_client = None
-        self.app_wrapper_name = self.config.name
-        if is_notebook():
+        if is_notebook() and config != None:
```

Comment:
Since we are returning early, we can remove the `else` and execute the code alongside the rest.

We can also remove the `config != None` check as it had already been checked.